packages/fuel-indexer-api-server/src/api.rs

```diff
@@ -103,19 +103,23 @@ impl IntoResponse for ApiError {
 pub struct GraphQlApi;
 
 impl GraphQlApi {
-    pub async fn run(config: IndexerConfig, tx: Option<Sender<ServiceRequest>>) {
-        let sm = SchemaManager::new(&config.database.to_string())
-            .await
-            .expect("SchemaManager create failed");
+    pub async fn run(
+        config: IndexerConfig,
+        connection_pool: Option<IndexerConnectionPool>,
+        tx: Option<Sender<ServiceRequest>>,
+    ) {
```

Comment:
What is the case where we don't pass a pool, but need to create a pool? The `GraphQLApi` can't really do anything w/o a DB connection so we can change `Option<IndexerConnectionPool>` to `IndexerConnectionPool`

Reply:
Currently? No case.

I'll change it. 👍🏽